### ISSUE TITLE
Option for disabling default language build

### ISSUE THREAD

**Issue by davidejones** [gohugoio/hugo#6377]
So in hugo we have the disable languages configuration option and the environment variable

```
disableLanguages = ["fr", "ja"]`
HUGO_DISABLELANGUAGES="fr ja" hugo
```

## Scenario

I have a website with multiple languages. I'm currently using the environment variable to build each language site separately in different jobs during our CI/CD. So I can build only on certain criteria.

### Job 1 - building the (default) english site only
`HUGO_DISABLELANGUAGES="fr ja" hugo`

### Job 2 - building the french site only
`HUGO_DISABLELANGUAGES="en ja" hugo`

### Job 3 - building the japanese site only
`HUGO_DISABLELANGUAGES="en fr" hugo`

The commands with `en` don't actually work as hugo will not allow me to disable the default language. So instead i have to remove `en` but that of course means that..

job 1 builds english
job 2 builds french & english
job 3 builds japanese & english

Which makes my approach not as optimal as it could be.

## Proposal

It seems to make sense to enforce a default language in Hugo so I'm not so sure if changing that any further is the right approach. However it would be a useful feature if the default language could be disabled just for separate builds, so that in my example each job only builds only the language it needs.

**Comment by bep** [comment]
I'm not sure as to the "why" here, but when you `disable` a language, that language does not exist (in translations, languages etc.), which does not make sense if your goal is to partition the build (to make it faster?). The motivation behind this feature was to make it easier to translate to new languages and then toggle it on when "good enough".

What I suspect you want is to filter the rendering job in Hugo to just include 1 language (or whatever).

I guess that would be useful, but I will wait a little. I'm doing a lot of thinking in this area (re. my work on "source plugins", pages from data etc.).

**Comment by davidejones** [comment]
> What I suspect you want is to filter the rendering job in Hugo to just include 1 language (or whatever).

Yes exactly, apologies if i didn't explain that very well.

> I guess that would be useful, but I will wait a little. I'm doing a lot of thinking in this area (re. my work on "source plugins", pages from data etc.).

That would be great! If you are waiting a while Is there anything I could contribute to make this happen sooner? or would you not accept a patch if you plan to rework that area?

**Comment by sergio-carlavilla** [comment]
@bep any news with this?
In the FreeBSD project we are using Hugo and it would save us a lot of power if we could disable the default language in some situations :)

**Comment by jmooring** [comment]
The underlying use case was addressed in v0.124.0 with segmented builds.
<https://gohugo.io/getting-started/configuration/#configure-segments>
